avocado/core/varianter.py

```diff
@@ -377,7 +377,7 @@ def _skip_new_data_check(self, function, args):
         """
         if self.is_parsed():
             if self.ignore_new_data:
-                return
+                return True
```

Comment:
Well it was on purpose as in heading I thought it'd look better, but I'll change it in the next v.